tests/scenarios/template-tag-codemod-test.ts

```diff
@@ -34,6 +34,21 @@ tsAppScenarios
             return result;
           }
         `,
+        'custom-resolver.mjs': `
+          export default async function customResolver(path) {
+            if (path === '@embroider/virtual/components/fancy-ice-cream') {
+              return 'bar/really-exists/foo';
+            }
+          }
+        `,
+        'another-resolver.mjs': `
+          export default async function(path, filename, resolve) {
+            if (path === '@embroider/virtual/components/phone-booth') {
+              path = '@embroider/virtual/components/message-box';
+            }
+            return await resolve(path, filename);
```

Comment:
So micro point on this test, the intention of this API is to allow you resolve **first** and if it gives you an answer that you don't agree with then you can change it before returning. This doesn't really matter because the test clearly shows that the resolve function works 👍